python/ql/src/experimental/Security/CWE-611/SimpleXmlRpcServer.ql

```diff
@@ -0,0 +1,27 @@
+/**
+ * @name SimpleXMLRPCServer DoS vulnerability
+ * @description SimpleXMLRPCServer is vulnerable to DoS attacks from untrusted user input
+ * @kind path-problem
+ * @problem.severity warning
+ * @precision high
+ * @id py/simple-xml-rpc-server
+ * @tags security
+ *       external/cwe/cwe-776
+ */
+
+private import python
+private import semmle.python.dataflow.new.DataFlow
+private import semmle.python.Concepts
+private import experimental.semmle.python.Concepts
+private import semmle.python.ApiGraphs
+
+from DataFlow::CallCfgNode call, string kinds
+where
+  call = API::moduleImport("xmlrpc").getMember("server").getMember("SimpleXMLRPCServer").getACall() and
```

Comment:
```suggestion
  call = API::moduleImport("xmlrpc").getMember("server").getMember("SimpleXMLRPCServer").getACall() and
  exists(call.getAMethodCall("register_function")) and
```

This ensures the server is actually being used to handle stuff.

Reply:
I have two objections:

1. there are many ways to add handlers, see [example docs](https://docs.python.org/3.10/library/xmlrpc.server.html#simplexmlrpcserver-example)
2. Why would you create a `SimpleXMLRPCServer` if you're not going to use it? 

Chances are that our logic for detecting whether the server is actually used is going to be imprecise, and we would not alert in some cases. So I would rather keep the check simple like it is already

Reply:
Given the objections, I strongly agree 👍